src/category_theory/fully_faithful.lean

```diff
@@ -81,6 +80,71 @@ variables (F : C ⥤ D) (G : D ⥤ E)
 
 instance faithful.comp [faithful F] [faithful G] : faithful (F ⋙ G) :=
 { injectivity' := λ _ _ _ _ p, F.injectivity (G.injectivity p) }
+
+lemma faithful.of_comp [faithful $ F ⋙ G] : faithful F :=
+{ injectivity' := λ X Y, (F ⋙ G).injectivity.of_comp }
+
+variables {F G}
+
+lemma faithful.of_comp_eq {H : C ⥤ E} [ℋ : faithful H] (h : F ⋙ G = H) : faithful F :=
+@faithful.of_comp _ _ _ _ _ _ F G (h.symm ▸ ℋ)
+
+alias faithful.of_comp_eq ← eq.faithful_of_comp
+
+variables (F G)
+
+/-- “Divide” a functor by a faithful functor. -/
+protected def faithful.div (F : C ⥤ E) (G : D ⥤ E) [faithful G]
+  (obj : C → D) (h_obj : ∀ X, G.obj (obj X) = F.obj X)
+  (map : ∀ {X Y}, (X ⟶ Y) → (obj X ⟶ obj Y))
+  (h_map : ∀ {X Y} {f : X ⟶ Y}, G.map (map f) == F.map f) :
+  C ⥤ D :=
+{ obj := obj,
+  map := @map,
+  map_id' :=
+  begin
+    assume X,
+    apply G.injectivity,
+    apply eq_of_heq,
+    transitivity F.map (𝟙 X), from h_map,
+    rw [F.map_id, G.map_id, h_obj X]
+  end,
+  map_comp' :=
+  begin
+    assume X Y Z f g,
+    apply G.injectivity,
+    apply eq_of_heq,
+    transitivity F.map (f ≫ g), from h_map,
+    rw [F.map_comp, G.map_comp],
+    congr' 1;
+      try { exact (h_obj _).symm };
+      exact h_map.symm
+  end }
+
+lemma faithful.div_comp (F : C ⥤ E) [faithful F] (G : D ⥤ E) [faithful G]
+  (obj : C → D) (h_obj : ∀ X, G.obj (obj X) = F.obj X)
+  (map : ∀ {X Y}, (X ⟶ Y) → (obj X ⟶ obj Y))
+  (h_map : ∀ {X Y} {f : X ⟶ Y}, G.map (map f) == F.map f) :
+  (faithful.div F G obj @h_obj @map @h_map) ⋙ G = F :=
```

Comment:
@semorrison How evil is this equality of functors?

Reply:
This one I'm still a bit more scared of, but nothing seems to go wrong, so ...